app/src/main/java/com/google/android/gms/samples/wallet/PaymentsUtil.java

```diff
@@ -152,8 +152,12 @@ private static PaymentDataRequest createPaymentDataRequest(TransactionInfo trans
      */
     public static Task<Boolean> isReadyToPay(PaymentsClient client) {
         IsReadyToPayRequest.Builder request = IsReadyToPayRequest.newBuilder();
-        for (Integer allowedMethod : Constants.SUPPORTED_METHODS) {
-            request.addAllowedPaymentMethod(allowedMethod);
+        try {
+            for (Integer allowedMethod : Constants.SUPPORTED_METHODS) {
+                request.addAllowedPaymentMethod(allowedMethod);
+            }
+        } catch (ExceptionInInitializerError e) {
```

Comment:
Can you explain why this catch block is necessary?  Also prefer `Log.w` to printing stack traces.

Reply:
For example if you don't delete

    static {
        if (true) {
            throw new RuntimeException("[REMOVE ME] Please edit the Constants.java file per the"
                    + " instructions inside before trying to run this sample.");
        }
    }

you'll get

    12-19 07:59:40.192 26182-26182/com.google.android.gms.samples.wallet E/AndroidRuntime: FATAL EXCEPTION: main
                                                                                       Process: com.google.android.gms.samples.wallet, PID: 26182
                                                                                       java.lang.ExceptionInInitializerError
                                                                                           at com.google.android.gms.samples.wallet.PaymentsUtil.isReadyToPay(PaymentsUtil.java:158)
                                                                                           at com.google.android.gms.samples.wallet.CheckoutActivity.checkIsReadyToPay(CheckoutActivity.java:78)
                                                                                           at com.google.android.gms.samples.wallet.CheckoutActivity.onCreate(CheckoutActivity.java:72)
                                                                                           at android.app.Activity.performCreate(Activity.java:6285)
                                                                                           at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108)
                                                                                           at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2417)
                                                                                           at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2524)
                                                                                           at android.app.ActivityThread.access$900(ActivityThread.java:154)
                                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1391)
                                                                                           at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                           at android.os.Looper.loop(Looper.java:234)
                                                                                           at android.app.ActivityThread.main(ActivityThread.java:5526)
                                                                                           at java.lang.reflect.Method.invoke(Native Method)
                                                                                           at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                                                        Caused by: java.lang.RuntimeException: [REMOVE ME] Please edit the Constants.java file per the instructions inside before trying to run this sample.
                                                                                           at com.google.android.gms.samples.wallet.Constants.<clinit>(Constants.java:44)
                                                                                           at com.google.android.gms.samples.wallet.PaymentsUtil.isReadyToPay(PaymentsUtil.java:158) 
                                                                                           at com.google.android.gms.samples.wallet.CheckoutActivity.checkIsReadyToPay(CheckoutActivity.java:78) 
                                                                                           at com.google.android.gms.samples.wallet.CheckoutActivity.onCreate(CheckoutActivity.java:72) 
                                                                                           at android.app.Activity.performCreate(Activity.java:6285) 
                                                                                           at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108) 
                                                                                           at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2417) 
                                                                                           at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2524) 
                                                                                           at android.app.ActivityThread.access$900(ActivityThread.java:154) 
                                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1391) 
                                                                                           at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                                           at android.os.Looper.loop(Looper.java:234) 
                                                                                           at android.app.ActivityThread.main(ActivityThread.java:5526) 
                                                                                           at java.lang.reflect.Method.invoke(Native Method) 
                                                                                           at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
                                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 

Reply:
@AlexanderLS I think that's what we want, to crash the app if developers have not set things up properly since many devs don't watch the logs.  This would turn it only into a logged warning right?

Reply:
Indeed, the reason for that exception in the static block is because we *really* want to ensure people read through and edited Constants.java before using the sample.

If you feel the wording around this could be improved, please do let us know - catching `ExceptionInInitializerError` is more harm than good.

Reply:
Indeed, the reason for that exception in the static block is because we *really* want to ensure people read through and edited Constants.java before using the sample.

If you feel the wording around this could be improved, please do let us know - catching `ExceptionInInitializerError` is more harm than good.